crates/handlers/src/graphql/mutations/user.rs

```diff
@@ -766,4 +860,154 @@ impl UserMutations {
+        }
+
+        if input.password.is_empty()
+            || !password_manager.is_password_complex_enough(&input.password)?
+        {
+            return RegisterUserStatus::InvalidPassword.into();
+        }
+
+        // If the site has terms of service, the user must accept them
+        if site_config.tos_uri.is_some() && !input.accept_terms {
+            return RegisterUserStatus::MustAcceptTerms.into();
+        }
+
+        let res = policy
+            .evaluate_register(&input.username, &input.email)
+            .await?;
+
+        if !res.violations.is_empty() {
+            let mut email_violations = Vec::new();
+            let mut username_violations = Vec::new();
+            let mut misc_violations = Vec::new();
+            for violation in res.violations {
+                match violation.field.as_deref() {
+                    Some("email") => {
+                        email_violations.push(violation.msg);
+                    }
+                    Some("username") => {
+                        username_violations.push(violation.msg);
+                    }
+                    _ => {
+                        misc_violations.push(violation.msg);
+                    }
+                }
+            }
+
+            return Ok(RegisterUserPayload {
+                status: RegisterUserStatus::PolicyViolation,
+                misc_violations,
+                username_violations,
+                email_violations,
+            });
+        }
+
+        let user = repo.user().add(&mut rng, &clock, input.username).await?;
+
+        if let Some(tos_uri) = &site_config.tos_uri {
+            repo.user_terms()
+                .accept_terms(&mut rng, &clock, &user, tos_uri.clone())
+                .await?;
+        }
+
+        let password = Zeroizing::new(input.password.into_bytes());
+        let (version, hashed_password) = password_manager.hash(&mut rng, password).await?;
+        let user_password = repo
+            .user_password()
+            .add(&mut rng, &clock, &user, version, hashed_password, None)
+            .await?;
+
+        let user_email = repo
+            .user_email()
+            .add(&mut rng, &clock, &user, input.email)
+            .await?;
+
+        let session = repo
+            .browser_session()
+            .add(&mut rng, &clock, &user, ctx.user_agent().cloned())
+            .await?;
+
+        repo.browser_session()
+            .authenticate_with_password(&mut rng, &clock, &session, &user_password)
+            .await?;
+
+        repo.job()
+            .schedule_job(
+                VerifyEmailJob::new(&user_email).with_language(ctx.preferred_locale().to_string()),
+            )
+            .await?;
+
+        repo.job()
+            .schedule_job(ProvisionUserJob::new(&user))
+            .await?;
+
+        repo.save().await?;
+
+        activity_tracker
+            .record_browser_session(&clock, &session)
+            .await;
+
+        RegisterUserStatus::Allowed.into()
```

Comment:
Not sure if possible, but this should set the session cookie, else the user won't be authenticated after this

Reply:
not exactly proud of it, but have added a workaround for this now